lib/diaspora_federation/schemas/federation_entities.json

```diff
@@ -0,0 +1,366 @@
+          "properties": {
+            "guid": { "type": "string" },
+            "public": { "type": "boolean" },
+            "created_at": { "type": "string" },
+            "provider_display_name" : { "type": "string" }
+          },
+          "required": [
+            "guid", "public", "created_at", "provider_display_name"
+          ]
+        },
+        "required": [
+          "entity_type", "entity_data"
+        ]
+      }
+    },
+
+    "status_message": {
+      "allOf": [
+        {"$ref": "#/definitions/post"},
+        {
+          "type": "object",
+          "properties": {
+            "entity_type": {
+              "type": "string",
+              "pattern": "^status_message$"
+            },
+
+            "entity_data": {
+              "type": "object",
+              "properties": {
+                "text": { "type": "string" },
+
+                "location": {
+                  "oneOf": [
+                    { "$ref": "#/definitions/location" },
+                    { "type": "null" }
+                  ]
+                },
+
+                "poll": {
+                  "oneOf": [
+                    { "$ref": "#/definitions/poll" },
+                    { "type": "null" }
+                  ]
+                },
+
+                "photos": {
+                  "type": ["array", "null"],
+                  "items": { "$ref": "#/definitions/photo" }
+                }
+              },
+
+              "required": ["text", "location", "poll", "photos"]
+            }
+          }
+        }
+      ]
+    },
+
+    "reshare": {
+      "allOf": [
+        {"$ref": "#/definitions/post"},
+        {
+          "type": "object",
+          "properties": {
+            "entity_type": {
+              "type": "string",
+              "pattern": "^reshare$"
+            },
+
+            "entity_data": {
+              "type": "object",
+              "properties": {
+                "root_author": {"type": "string"},
+                "root_guid": {"type": "string"}
+              },
+
+              "required": ["root_author", "root_guid"]
+            }
+          }
+        }
+      ]
+    },
+
+    "profile": {
+      "type": "object",
+      "properties": {
+        "entity_type": {
+          "type": "string",
+          "pattern": "^profile$"
+        },
+        "entity_data": {
+          "type": "object",
+          "properties": {
+            "first_name": { "type": ["string", "null"] },
+            "last_name": { "type": ["string", "null"] },
+            "gender": { "type": ["string", "null"] },
+            "bio": { "type": ["string", "null"] },
+            "birthday": { "type": ["string", "null"] },
+            "location": { "type": ["string", "null"] },
+            "image_url": { "type": ["string", "null"] },
+            "author": { "type": "string" }
+          },
+          "required": [
+            "first_name", "last_name", "gender", "bio", "birthday", "location", "image_url", "author"
```

Comment:
The only field required here is `author`, see https://diaspora.github.io/diaspora_federation/entities/profile.html